extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentMover.java

```diff
@@ -133,18 +134,33 @@ public Void call() throws Exception
                     s3Object.getStorageClass().equals(S3Object.STORAGE_CLASS_GLACIER)) {
                   log.warn("Cannot move file[s3://%s/%s] of storage class glacier, skipping.", s3Bucket, s3Path);
                 } else {
-                  log.info(
-                      "Moving file[s3://%s/%s] to [s3://%s/%s]",
-                      s3Bucket,
+                  final String copyMsg = StringUtils.safeFormat(
+                      "[s3://%s/%s] to [s3://%s/%s]", s3Bucket,
                       s3Path,
                       targetS3Bucket,
                       targetS3Path
                   );
+                  log.info(
+                      "Moving file %s",
+                      copyMsg
+                  );
                   final S3Object target = new S3Object(targetS3Path);
                   if (!config.getDisableAcl()) {
                     target.setAcl(GSAccessControlList.REST_CANNED_BUCKET_OWNER_FULL_CONTROL);
                   }
-                  s3Client.moveObject(s3Bucket, s3Path, targetS3Bucket, target, false);
+                  final Map<String, Object> copyResult = s3Client.moveObject(
+                      s3Bucket,
+                      s3Path,
+                      targetS3Bucket,
+                      target,
+                      false
+                  );
+                  if (copyResult != null && copyResult.containsKey("DeleteException")) {
+                    log.error("Error Deleting data after copy %s: %s", copyMsg, copyResult);
+                    // Maybe retry deleting here?
```

Comment:
Retry the delete? I had considered that, but wasn't 100% sure if that was desired. So I opted for "maintain current behavior but log more"